### PR TITLE
Adding git secrets to prevent accidental commits

### PR DESCRIPTION
**Pull request by kss39**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+repos:
+-   repo: https://github.com/awslabs/git-secrets
+    rev: 80230afa8c8bdeac766a0fece36f95ffaa0be778
```
Comment (zaro0508) [inline .pre-commit-config.yaml]: i believe you only need first 7 numbers of the SHA?  Also wondering why you don't set it to a released tag?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    hooks:
+        - id: git-secrets
```
Comment (zaro0508) [inline .pre-commit-config.yaml]: hmm, does this work?  don't we need to run the git-secrets script like the one in https://github.com/awslabs/git-secrets/blob/master/.pre-commit-hooks.yaml#L4 ?

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (kss39) [reply .pre-commit-config.yaml]: Hi,

Actually, the most recent released tag (1.3.0) does not support the pre_commit_hook structure. It was introduced after this release (this commit: b4a3208)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (zaro0508) [reply .pre-commit-config.yaml]: just verified myself.  this, as-is, does indeed work.

**Review by zaro0508 — CHANGES_REQUESTED** [review]
since running pre-commit locally (on your own dev machine) is voluntary i would advise you to also make this run as part of the CI process.  That means you should setup pre-commit to run on every PR and merge.  You can find examples of this setup just about all of the `Sage-Bionetworks/*-infra` repos that have a .pre-commit-config.yaml file.

**Comment by cascadianblue** [comment]
Have you considered adding any other pre-commit hooks (i.e. linters, etc.)? I'm betting that a lot of committers will forget to run/install pre-commit locally (especially if developers work across multiple environments) even if you add it to your documentation. Adding more other hooks might make it more likely for committers to accidentally trip up the CI-side pre-commit check (e.g. with forgetting a newline at the end of a Python file) and force them to install pre-commit locally. Make sense?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (boyleconnor) [inline .pre-commit-config.yaml]: Have you considered adding any other pre-commit hooks (i.e. linters, etc.)? I'm betting that a lot of committers will forget to run/install pre-commit locally (especially if developers work across multiple environments) even if you add it to your documentation. Adding more other hooks might make it more likely for committers to accidentally trip up the CI-side pre-commit check (e.g. with forgetting a newline at the end of a Python file) and force them to install pre-commit locally. Make sense?

**Review by boyleconnor — CHANGES_REQUESTED** [review]

**Review by DwayneJengSage — APPROVED** [review]

**Comment by alxdarksage** [comment]
Please add something to the README about any actions developers need to take to ensure this is doing its job. Anything we need to install or run?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (kss39) [reply .pre-commit-config.yaml]: Thank you, yes, I will add more checks for it.

**Comment by kss39** [comment]
> Please add something to the README about any actions developers need to take to ensure this is doing its job. Anything we need to install or run?

Yes, as far as I tested, all we need to do is `pip install pre-commit` and
`pre-commit install` in BridgeServer2/ folder.

As for where to put the README, how about we put it on the wiki for developer bootstrapping? https://github.com/Sage-Bionetworks/BridgeServer2/wiki/bootstrap

I double checked, this pre-commit checks also works with IDE GUIs, say that if I committed an AWS key using IntelliJ's GUI commit window (instead of command line), IntelliJ will fail the commit process and raise an error.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+repos:
+-   repo: https://github.com/awslabs/git-secrets
+    rev: 80230af
+    hooks:
+        -   id: git-secrets
+-   repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (boyleconnor) [inline .pre-commit-config.yaml:6]: Might be good to include some general-purpose Java linting hooks, e.g. one of [these](https://github.com/gherynos/pre-commit-java).

**Review by boyleconnor — APPROVED** [review]

**Review by zaro0508 — APPROVED** [review]

**Pull request closed without merging**